sunpy/net/dataretriever/sources/goes.py

```diff
@@ -128,17 +123,20 @@ def search(self, *args, **kwargs):
             metalist = self._get_metalist(matchdict)
         return QueryResponse(metalist, client=self)
 
-    def _get_metalist_fn(self, matchdict, baseurl, pattern):
+    def _get_metalist_fn(self, matchdict, pattern, **kwargs):
         """
         Function to help get list of OrderedDicts.
         """
         metalist = []
-        scraper = Scraper(baseurl, regex=True)
+        # Prevent parse-format pattern variables from being interpreted as .format() placeholders.
+        if "{{" not in pattern:
+            pattern = pattern.replace('{', '{{').replace('}', '}}')
```

Comment:
Will this be needed? If the wrong parse holder was added, I think we should error no?

Reply:
This is no longer needed yes, I initially added this because when the pattern reached this stage, `format()` had already been called on it once for some clients. But instead of calling `format()` on pattern while passing it to the function, I had parameters passed as kwargs. So yeah I just double-checked, we can remove this